src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -462,9 +461,10 @@ public boolean isModelInGraveyard(String modelId) {
         public void delete(String modelId, ActionListener<DeleteModelResponse> listener) {
             // If the index is not created, there is no need to delete the model
             if (!isCreated()) {
-                logger.error("Cannot delete model \"" + modelId + "\". Model index " + MODEL_INDEX_NAME + "does not exist.");
-                String errorMessage = String.format("Cannot delete model \"%s\". Model index does not exist", modelId);
-                listener.onResponse(new DeleteModelResponse(modelId, FAILED, errorMessage));
+                String errorMessage = String.format("Cannot delete model [%s]. Model index [%s] does not exist", modelId, MODEL_INDEX_NAME);
+                Exception e = new ResourceNotFoundException(errorMessage);
+                logger.error(e);
```

Comment:
This is really interesting use of exception. In general we log the exception trace so that we know from where exception came. but here this is not the case.

If we put a log.error message here will that not be sufficient?

Reply:
We also need exception instance to pass it to the listener.onFailure, I guess that's the main reason why we create it

Reply:
> In general we log the exception trace so that we know from where exception came. but here this is not the case.

Im not sure what you mean by this. I wanted to log the error here so that we can see the complete stack trace. We can also just log the errorMessage, but I felt adding additional information about the exception would be helpful.

Reply:
What information you are adding here? I see log can do the job.




> Im not sure what you mean by this.

What I mean was why not log the error message in when `listener.onFailure(e);` is executed, which is the line just below this code.

Reply:
Got it, I updated.